draft-ietf-quic-ack-frequency.md

```diff
@@ -382,15 +382,22 @@ Unreported Missing:
   Missing that have not yet been received.
 
 An endpoint that receives an ACK_FREQUENCY frame with a non-zero Reordering
-Threshold value SHOULD send an immediate ACK when the gap
-between the smallest Unreported Missing packet and the Largest Unacked is greater
-than or equal to the Reordering Threshold value. Sending this additional ACK will
-reset the max_ack_delay timer and Ack-Eliciting Threshold counter (as any ACK
-would do).
-
-When an ack-eliciting packet is received with a packet number less than
-Largest Acked, this still triggers an immediate acknowledgement in an effort to
-avoid the packet being spuriously declared lost.
+Threshold value SHOULD send an immediate ACK whenever it receives an ack-eliciting,
+out-of order packet whose packet number is outside the reordering window of the peer,
+i.e. when
+* the difference between the smallest Unreported Missing packet and the
+  Largest Unacked packet is greater than or equal to the Reordering
+  Threshold value; or
+* the received packet number is less than or equal to `Largest Acked - Reordering Threshold`.
+
+The first condition triggers an ACK as soon as the reordering threshold is reached and
+a packet can be declared lost at the sender. The second condition addresses packets
+that have been received later, out of order and are already spuriously declared lost
+by the sender. This enables the sender to detect spurious retransmissions and revert
+the congestion control status accordingly.
+
+Sending such an additional ACK resets the max_ack_delay timer and the
+Ack-Eliciting Threshold counter, as any ACK would.
```

Comment:
```suggestion
Ack-Eliciting Threshold counter, as any ACK would.

In order to avoid unnecessary immediate acknowledgements, implementations SHOULD implement adaptive packet threshold loss detection and communicate the increased Reordering Threshold value to the receiver.
```